src/admin.js

```diff
@@ -1,14 +1,26 @@
 /* global window, fetch */
 /* eslint no-labels: ['error', { 'allowLoop': true }] */
 import 'fetch-everywhere';
-import StitchClient from './client';
+import { StitchClient } from './client';
 import ADMIN_CLIENT_TYPE from './common';
 import { ADMIN_CLIENT_CODEC } from './auth/common';
 
 const v2 = 2;
 const v3 = 3;
 
-export default class Admin extends StitchClient {
+export class StitchAdminFactory {
+  constructor() {
+    throw new StitchError('StitchAdminClient can only be made from the StitchAdminFactory.create function');
+  }
+
+  static create(baseUrl) {
+    return new Promise((resolve, reject) => {
+      resolve(new StitchAdminClient(baseUrl)._willInitialize);
+    });
+  }
+}
+
+export class StitchAdminClient extends StitchClient {
```

Comment:
[minor] We shouldn't be exporting the client. Only the `StitchAdminFactory` should be exported (can be default as well)